metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -91,6 +91,31 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
             else:
                 self.report.report_dropped(t)
 
+    def get_schema_str_replace_confluent_ref_avro(
+        self, schema: Schema, schema_seen: Dict = None
+    ) -> str:
+        if not schema.references:
+            return schema.schema_str
+
+        if schema_seen is None:
+            schema_seen = dict()
+        schema_str = schema.schema_str
+        for schema_ref in schema.references:
+            ref_subject = schema_ref["subject"]
+            ref_name = schema_ref["name"]
+            if ref_subject in schema_seen:
```

Comment:
Looks like `schema_seen` is being used more like a `set` than a `dict`, and is only part of the formal params to facilitate recursive calls. Consider eliminating it from the formal params and use an inner/helper function to facilitate the recursion.